src/helper.js

```diff
@@ -271,12 +272,24 @@ function passwordCompare(password, expected) {
 	return bcrypt.compare(password, expected);
 }
 
-function getDefaultNick() {
-	if (!this.config.defaults.nick) {
+function formatDefaultNick(format) {
+	if (!format) {
 		return "thelounge";
 	}
 
-	return this.config.defaults.nick.replace(/%/g, () => Math.floor(Math.random() * 10));
+	return format.replace(/%/g, () => Math.floor(Math.random() * 10));
+}
+
+function getDefaultNetworks() {
+	if (this.config.defaults === undefined) {
+		return [];
+	}
+
+	if (Array.isArray(this.config.defaults)) {
+		return this.config.defaults;
+	}
+
+	return [this.config.defaults];
```

Comment:
Can we also add a log warning here that the object format is deprecated?

Something like https://github.com/thelounge/thelounge/pull/4472/files#diff-85b18f65b01cea7762ab044d7df73c7d3e6de0c9e3502cce8540cad8f2534039R283

Reply:
Oh yes, of course. I don't see what your link is supposed to point to, so I improvised. The output format looks like this:

```
2022-02-14 18:32:26 [WARN] Key "defaults" should now be an array of networks, please update your config. It should look like this:
defaults: [
	{
		"name": "Libera.Chat",
		"host": "irc.Libera.Chat",
		"port": 6697,
		"password": "",
		"tls": true,
		"rejectUnauthorized": true,
		"nick": "thelounge%%",
		"username": "thelounge",
		"realname": "The Lounge User",
		"join": "#progval",
		"leaveMessage": ""
	}
],
```

I can change it, of course

Reply:
Ah, I used the wrong link. This message works, but I'd include "support for the old object format will be removed in a future version". Not sure we need to print the defaults, maybe a link to the docs?

Reply:
Done. I pluralized the anchor in the link, as I expect the title to be made plural when releasing the next version.

Reply:
Yep, the docs are auto-generated from the config file comments so you're change here will change the anchor:
https://github.com/thelounge/thelounge/pull/4472/files#diff-2a8a9985100687b211e3cf1123b40796d8091ac34db900f6683a81ef06046219R207